### PR TITLE
favorites related queries, routes, ejs templates(placeholder)

### PR DESCRIPTION
**Pull request by haijoon2**
## favorites queries

1. created a query for getting all the favorites by given userId
2. created a query for favoring an item by itemId & userId
3. created a query for unfavoring an item by favoriteId & userId
---
## favorites routes
1. GET /favorites => getting all the favorites by a logged in user
2. POST /favorites => creating a favorite by given itemId by a logged in user
3. DELETE /favorites/:favoriteId => deleting a favorite by given favoriteId by a logged in user
---
## favorites ejs templates
* placeholder for displaying 4 favorite items per row
Screenshot 2023-08-20 at 3 40 40 PM
- I didn't do any designs yet because we can share the item card rendering function.
I will connect it later once the item card rendering function is done.

---
### FOR TESTING
- run the app and you can access the page to [this url](http://localhost:8080/favorites)

response example
Screenshot 2023-08-20 at 3 50 49 PM

### CODE REVIEW

File: db/queries/favorites.js [hunk 1]
```diff
@@ -1,18 +1,73 @@
...
+ * @property {number} price - the price of the item in cents
+ * @property {number} item_id - the id of the item
+ */
+
+/**
+ * Get favorites from the database by user id.
+ *
+ * @param {number} userId - the id of the user
+ * @return {Promise<{ favorites?: Favorite[], error?: any }>} A promise to the favorites.
+ */
```
Comment (haijoon2) [inline db/queries/favorites.js:16]: I saw jsdoc was used so I followed it.

File: views/favorites.ejs [hunk 2]
```diff
@@ -7,21 +7,21 @@
...
 
-  <div class="card" style="width: 18rem;">
-    <img class="card-img-top" src="..." alt="Card image cap">
-    <div class="card-body">
-      <h5 class="card-title">Card title</h5>
-      <p class="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
-      <a href="#" class="btn btn-primary">Go somewhere</a>
-    </div>
-  </div>
-
```
Comment (haijoon2) [inline views/favorites.ejs:24]: I deleted it since it was a placeholder

File: db/seeds/01_users_seeds.sql [hunk 3]
```diff
@@ -1,30 +1,30 @@
-insert into users (id, name, email, phone_number, password) values (1, 'Ethelyn Scamwell', 'escamwell0@uiuc.edu', '531-607-6170', 'jB7#&nm1*MAuJ&N');
```
Comment (haijoon2) [inline db/seeds/01_users_seeds.sql:1]: I kept having troubles when testing favorite/unfavorite routes, so I reached out for help and below were the answers.
"we defined ids serial so postgres automatically gives IDs and counts them. 
If we give ids manually when seeding, it erros out when we try to create an entry."

so I removed ids from seeds.

**Review by Sunnyhundal — APPROVED** [review]
Looks good to me

**Pull request merged**